uninstall.php

```diff
@@ -7,6 +7,15 @@
 
 defined( 'WP_UNINSTALL_PLUGIN' ) || exit;
 
+// Don't delete table data if core WooCommerce contains WooCommerce Admin.
+// This could alternatively check for a specific WooCommerce version rather than a file.
+if (
+    defined( 'WC_ABSPATH' ) &&
```

Comment:
Nice tactic here!

I think it may be an edge case (if at all), but does `uninstall.php` run after the `plugins_loaded`?  Is there a case where `WC_ABSPATH ` is not yet defined since wc-admin loaded first?

Reply:
`uninstall.php` is invoked in `uninstall_plugin()` through `delete_plugins()` and always occurs after `plugins_loaded`.

See:
https://github.com/WordPress/WordPress/blob/a5e52e04ee5f4943b3883e7857bdc28a7594bf68/wp-admin/plugins.php#L380

https://github.com/WordPress/WordPress/blob/f793398e4f26051a49abd03a084fb43d3c578c3b/wp-admin/includes/ajax-actions.php#L4541